Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2399 @@
+ /* Determine the number of CPCs represented in the array     */
+ /**************************************************************/
+ CPCs = JSON_getArrayDim(CPCArray)
+ if CPCs <= 0 then
+    return fatalError( '** Unable to retrieve number of array entries **' )
+
+ /********************************************************************/
+ /* Traverse the CPCs array to populate CPCsList.                    */
+ /* We use the REXX (1-based) idiom  but adjust for the differing    */
+ /* (0-based) idiom of the toolkit.                                  */
+ /********************************************************************/
+ say 'Processing information for '||CPCs||' CPC(s)... searching for LOCAL'
+ do i = 1 to CPCs
+    nextEntryHandle = JSON_getArrayEntry(CPCArray,i-1)
+    CPClocation = JSON_findValue(nextEntryHandle,"location",HWTJ_STRING_TYPE)
+
+    if CPClocation = 'local' then
+      do
+        say 'found LOCAL CPC'
+        if localCPCfound = TRUE then
+          do
+            return fatalError('** found two LOCAL CPCs **')
+          end
+        localCPCfound = TRUE
+
+        CPCuri=JSON_findValue(nextEntryHandle,"object-uri",HWTJ_STRING_TYPE)
+        if foundJSONValue(CPCuri) = FALSE then
+          do
+            return fatalError,
+            ('ERROR: found LOCAL CPC but it did not contain an object-uri')
+          end
+
+        CPCtargetName=JSON_findValue(nextEntryHandle,"target-name",,
+                                    HWTJ_STRING_TYPE)
+        if foundJSONValue(CPCtargetName) = FALSE then
+          do
+            return fatalError,
+            ('ERROR: found LOCAL CPC but it did not contain target-name')
+          end
+
+        CPCname=JSON_findValue(nextEntryHandle,"name",HWTJ_STRING_TYPE)
+        if foundJSONValue(CPCname) = FALSE then
+          do
+            return fatalError,
+            ('ERROR: found LOCAL CPC but it did not contain a name property')
+          end
+      end
+ end /* endloop thru the JSON LPARs array */
+
+if localCPCfound then
+  do
+    Say
+    Say 'Successfully obtained LOCAL CPC Info:'
+    Say '  name:'||CPCname
+    Say '  uri:'||CPCuri
+    Say '  target-name:'||CPCtargetName
+    Say
+    return 0
+  end
+
+
+/* if we're still here than something went wrong */
+return fatalError('ERROR - ** failed to obtain LOCAL CPC info **')
+
+/*******************************************************/
+/* Function:  GetCPCInfo                               */
+/*                                                     */
+/* Retrieve the uri, target name and SE version        */
+/* associated with CPCname                             */
+/* Prime CPCuri and CPCtargetName variables            */
+/* Verify SE version is z16 or higher                  */
+/*                                                     */
+/* Return 0 if successful, otherwise return -1         */
+/*******************************************************/
+GetCPCInfo:
```

Comment:
Is it possible to adjust GetCPCInfo to take a boolean indicating whether the user wishes to get the local CPC or a specific CPC info? Then we can simply remove GetLocalCPCInfo as it is no longer needed and would be duplicate code